### PR TITLE
Optimize processing memory usage

### PR DESCRIPTION
**Pull request by arnaud-morvan**
- Iterate over admindiv primary key identifier, retrieve and expunge records one by one in loop.
- Remove admindiv geometry simplification.

This modifications increase the processing time by approximatively 25% to save 700Mo of memory on global hazardsets.

### CODE REVIEW

File: thinkhazard_processing/processing.py [hunk 1]
```diff
@@ -174,81 +173,92 @@ def create_outputs(hazardset, layers, readers):
...
-            outputs.append(output)
-
-        percent = int(100.0 * current / total)
-        if percent % 10 == 0 and percent != last_percent:
-            logger.info('  ... processed {}%'.format(percent))
-            last_percent = percent
+            shape = to_shape(admindiv.geom)
 
-    return outputs
+            if not shape.intersects(bbox):
```
Comment (pgiraud) [inline thinkhazard_processing/processing.py]: Is it still needed? Isn't the filter in the database already doing the job?

File: thinkhazard_processing/processing.py [hunk 2]
```diff
@@ -174,81 +173,92 @@ def create_outputs(hazardset, layers, readers):
...
+                        readers, shape)
+
+            except Exception as e:
+                e.message = ("{}-{} raises an exception :\n{}"
+                             .format(admindiv.code, admindiv.name, e.message))
+                raise
+
+            # Create output record
+            if hazardlevel is not None:
+                # print '    hazardlevel :', output.hazardlevel.mnemonic
```
Comment (pgiraud) [inline thinkhazard_processing/processing.py]: Please remove this comment.

File: thinkhazard_processing/processing.py (hunk 1 above)
Comment (arnaud-morvan) [inline thinkhazard_processing/processing.py]: Yes, in case of global hazardsets, that we do not filter in database query.

Comment (arnaud-morvan) [inline thinkhazard_processing/processing.py]: I think that we can use filter in database query, now that administrative division are in 4326

**Comment by pgiraud** [comment]
Please merge

**Pull request merged**